### PR TITLE
vue3: Added share view

### PR DESCRIPTION
**Pull request by mustafapc19**
Ref: #3038 

Added
- Grid
- `Fields` to hide/show columns
- Integrated `filter`
- Integrated `sort`
- Integrated `filter groups`
- Integrate `more`
- Added password login
- Integrated hiding `more` option

Added initial works for form view. Pranav will be doing the rest

UI:
image
image
image
image
image
image
image

### CODE REVIEW

**Review by pranavxc — CHANGES_REQUESTED** [review]

File: packages/nc-gui-v2/composables/useSharedView.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import type { ColumnType, ExportTypes, FilterType, PaginatedType, SortType, TableType, ViewType } from 'nocodb-sdk'
+import { UITypes } from 'nocodb-sdk'
+import { useNuxtApp } from '#app'
+
+const nestedFilters = ref<(FilterType & { status?: 'update' | 'delete' | 'create'; parentId?: string })[]>([])
+const paginationData = ref<PaginatedType>({ page: 1, pageSize: 25 })
```
Comment (pranavxc) [inline packages/nc-gui-v2/composables/useSharedView.ts:6]: Why are all the refs outside the composable? If you want to share the state use `useState`.

**Comment by dstala** [comment]
- Observations `Shared Grid View`
- [ ] Entire shared screen is grayed out. Intended?
- [ ] Error on expanding cell for LTAR field
  
Screenshot 2022-08-15 at 12 43 48 PM (2)

- [ ] `Show System Fields` was disabled in Shared view (v1)
  
Screenshot 2022-08-15 at 12 48 01 PM

- [ ] Expand row- error; was disabled in v1
  
Screenshot 2022-08-15 at 12 50 36 PM

- [ ] Disable context-menu, check-box to select
  
Screenshot 2022-08-15 at 12 50 21 PM

- [ ] View name not displayed

- Verified
- [ ] Field hide / re-order
- [ ] Sort
- [ ] Filter
- [ ] Download files
- [ ] RO fields

**Review by dstala — CHANGES_REQUESTED** [review]
https://github.com/nocodb/nocodb/pull/3083

**Comment by mustafapc19** [comment]
@dstala Regarding some of the cell issues. Its due to cell not configured for read only mode. Which is out of scope for this PR.

And can we ignore the issues of this PR for now and merge?(Unless its breaking stuff) This pr is touching on a lot files and had a lot of conflicts last time. And seems to be in conflict again. i.e I think I did disable expanded row view but seems to slip back in during conflict resolution. Will send a PR again as a followup?

**Comment by dstala** [comment]
> @dstala Regarding some of the cell issues. Its due to cell not configured for read only mode. Which is out of scope for this PR.
> 
> And can we ignore the issues of this PR for now and merge?(Unless its breaking stuff) This pr is touching on a lot files and had a lot of conflicts last time. And seems to be in conflict again. i.e I think I did disable expanded row view but seems to slip back in during conflict resolution. Will send a PR again as a followup?

@mustafapc19 sure. We can merge these changes & work on the rest of the items in a new PR

**Review by dstala — APPROVED** [review]

**Comment by mustafapc19** [comment]
@bcakmakoglu While reviewing can you specify the deal breakers? Will only include them and include the rest of the reviews in a follow up PR.

File: packages/nc-gui-v2/components/shared-view/AskPassword.vue [hunk 2]
```diff
@@ -0,0 +1,49 @@
+<script setup lang="ts">
+import { message } from 'ant-design-vue'
+import { extractSdkResponseErrorMsg } from '~/utils'
+const props = defineProps<Props>()
+const emit = defineEmits(['update:modelValue'])
+interface Props {
+  modelValue: boolean
+}
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/shared-view/AskPassword.vue:8]: Interface goes above props

File: packages/nc-gui-v2/components/shared-view/AskPassword.vue [hunk 3]
```diff
@@ -0,0 +1,49 @@
+<script setup lang="ts">
+import { message } from 'ant-design-vue'
+import { extractSdkResponseErrorMsg } from '~/utils'
+const props = defineProps<Props>()
+const emit = defineEmits(['update:modelValue'])
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/shared-view/AskPassword.vue:5]: some space in between vars and imports would be nice

File: packages/nc-gui-v2/components/shared-view/Form.vue [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+const formRef = ref()
+
+watch(
+  () => formState,
+  () => {
+    console.log('formData', formState.value)
+  },
+)
+
+console.log(fields)
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/shared-view/Form.vue:39]: can be removed I assume

File: packages/nc-gui-v2/components/shared-view/Form.vue [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+</script>
+
+<template>
+  <div class="flex flex-col my-4 space-y-2 mx-32 items-center">
+    <div class="flex w-2/3 flex-col mt-10">
+      <div class="flex flex-col items-start px-14 py-8 bg-gray-50 rounded-md w-full">
+        <a-typography-title class="border-b-1 border-gray-100 w-full pb-3" :level="1">{{
+          sharedView.view.heading
+        }}</a-typography-title>
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/shared-view/Form.vue:48]: Formatting
Should be
```html
<a-typography-title class="border-b-1 border-gray-100 w-full pb-3" :level="1">
	{{ sharedView.view.heading }}
</a-typography-title>

File: packages/nc-gui-v2/components/smartsheet/Toolbar.vue [hunk 6]
```diff
@@ -1,7 +1,10 @@
 <script setup lang="ts">
 import { useSmartsheetStoreOrThrow } from '~/composables'
+import { IsPublicInj } from '~/context'
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet/Toolbar.vue:3]: Can be shorted to just one import line from `'#imports'`

File: packages/nc-gui-v2/composables/useViewData.ts [hunk 7]
```diff
@@ -5,6 +5,7 @@ import { useNuxtApp } from '#app'
 import { useProject } from '#imports'
 import { NOCO } from '~/lib'
 import { extractPkFromRow, extractSdkResponseErrorMsg } from '~/utils'
+import { IsPublicInj } from '~/context'
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/composables/useViewData.ts:8]: -> `'#imports'`

File: packages/nc-gui-v2/composables/useViewFilters.ts [hunk 8]
```diff
@@ -1,22 +1,55 @@
-import type { FilterType, ViewType } from 'nocodb-sdk'
+import type { ViewType } from 'nocodb-sdk'
 import type { ComputedRef, Ref } from 'vue'
 import { useNuxtApp, useUIPermission } from '#imports'
 import { useMetas } from '~/composables/useMetas'
+import { IsPublicInj, ReloadViewDataHookInj } from '~/context'
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/composables/useViewFilters.ts:5]: -> `'#imports'`

File: packages/nc-gui-v2/composables/useViewSorts.ts [hunk 9]
```diff
@@ -1,25 +1,53 @@
 import type { GalleryType, GridType, KanbanType, SortType } from 'nocodb-sdk'
 import type { Ref } from 'vue'
 import { useNuxtApp } from '#imports'
+import { IsPublicInj, ReloadViewDataHookInj } from '~/context'
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/composables/useViewSorts.ts:4]: -> `'#imports'`

File: packages/nc-gui-v2/pages/[projectType]/form/[viewId].vue [hunk 10]
```diff
@@ -0,0 +1,42 @@
+<script setup lang="ts">
+import type { Ref } from 'vue'
+import type { TableType } from 'nocodb-sdk/build/main'
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/pages/[projectType]/form/[viewId].vue:3]: -> `'nocodb-sdk'`

File: packages/nc-gui-v2/pages/[projectType]/form/[viewId].vue [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+import { ActiveViewInj, FieldsInj, IsPublicInj, MetaInj, ReloadViewDataHookInj } from '~/context'
+
+import { useRoute } from '#imports'
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/pages/[projectType]/form/[viewId].vue:7]: -> `'#imports'`

File: packages/nc-gui-v2/pages/[projectType]/form/[viewId].vue [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+definePageMeta({
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/pages/[projectType]/form/[viewId].vue:8]: spacing

File: packages/nc-gui-v2/pages/[projectType]/form/[viewId].vue [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+await loadSharedView(route.params.viewId as string)
+
+provide(ReloadViewDataHookInj, reloadEventHook)
+provide(MetaInj, meta)
+provide(ActiveViewInj, sharedView)
+provide(FieldsInj, formColumns)
+provide(IsPublicInj, ref(true))
+
+useProvideSmartsheetStore(sharedView as Ref<TableType>, meta)
+// useSmartsheetRowStore()
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/pages/[projectType]/form/[viewId].vue:26]: Can be removed

File: packages/nc-gui-v2/pages/[projectType]/view/[viewId].vue [hunk 14]
```diff
@@ -0,0 +1,33 @@
+<script setup lang="ts">
+import { ReadonlyInj, ReloadViewDataHookInj } from '~/context'
+
+import { useRoute } from '#imports'
+definePageMeta({
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/pages/[projectType]/view/[viewId].vue:5]: spacing

File: packages/nc-gui-v2/pages/[projectType]/view/[viewId].vue [hunk 15]
```diff
@@ -0,0 +1,33 @@
+<script setup lang="ts">
+import { ReadonlyInj, ReloadViewDataHookInj } from '~/context'
+
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/pages/[projectType]/view/[viewId].vue:3]: remove space

File: packages/nc-gui-v2/pages/[projectType]/view/[viewId].vue [hunk 16]
```diff
@@ -0,0 +1,33 @@
+<script setup lang="ts">
+import { ReadonlyInj, ReloadViewDataHookInj } from '~/context'
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/pages/[projectType]/view/[viewId].vue:2]: -> `'#imports'`

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

**Pull request merged**